tests/suites/secrets_iaas/juju.sh

```diff
@@ -121,6 +121,7 @@ run_user_secrets() {
 	# revision 1 should be pruned.
 	# revision 2 is still been used by the app, so it should not be pruned.
 	# revision 3 is the latest revision, so it should not be pruned.
+	sleep 5 # wait for prune to happen.
```

Comment:
Is there any way to do this without sleeping?

Reply:
We need to use a while loop with a max number of attempts like in other bash tests.

Reply:
Agreed